src/App.js

```diff
@@ -1,41 +1,60 @@
 import React, { Component, Fragment } from 'react'
 import Nav from './Nav'
-import NewCard from './NewCard'
+import Form from './Form'
 import CardList from './CardList'
 import CardListEmpty from './CardListEmpty'
+import parseHash from './parse-hash'
 
 export default class App extends Component {
   constructor(props) {
     super(props)
+    const { path, params } = parseHash(window.location.hash)
     const cards = localStorage.getItem('cards')
     this.state = {
-      path: window.location.hash.slice(1),
+      path,
+      params,
       cards: JSON.parse(cards) || []
     }
-    this.updateCardList = this.updateCardList.bind(this)
+    this.addToCardList = this.addToCardList.bind(this)
+    this.editCardList = this.editCardList.bind(this)
   }
 
   componentDidMount() {
     window.addEventListener('hashchange', () => {
-      const path = window.location.hash.slice(1)
-      this.setState({ path })
+      const { path, params } = parseHash(window.location.hash)
+      this.setState({ path, params })
     })
 
     window.addEventListener('beforeunload', () => {
       localStorage.setItem('cards', JSON.stringify(this.state.cards))
     })
   }
 
-  updateCardList(newCard) {
+  addToCardList(newCard) {
     const cards = [...this.state.cards, newCard]
-    this.setState({ cards: cards })
+    this.setState({ cards })
   }
 
-  renderNewCard() {
-    return <NewCard currentCards={this.state.cards} updateCardList={this.updateCardList} />
+  editCardList(newCard, editIndex) {
+    const cards = this.state.cards.map((card, index) => {
+      if (index === editIndex) return newCard
+      return card
+    })
+    this.setState({ cards })
+  }
+
+  renderForm(type) {
+    return <Form
+      type={type}
```

Comment:
Only tell the form what function to call on submit and what it's editing (if applicable).